test/test_c10d.py

```diff
@@ -970,6 +970,25 @@ def test_timeout_kwarg(self):
         # The barrier will now time output
         self.assertFalse(pg.barrier().wait())
 
+    @unittest.skip("Implementation of this functionality pending; see #14373")
+    def test_barrier_implies_wait(self):
+        store = c10d.FileStore(self.file.name, self.world_size)
+        pg = c10d.ProcessGroupGloo(store, self.rank, self.world_size)
```

Comment:
Just chime in here, I didn't remember the default number of ProcessGroupGloo threads. If it's not 2 or more, can you modify this to ensure that at least two threads are created, now that I saw your pending work changes rely on this.